src/configuration.rs

```diff
@@ -1,80 +1,281 @@
-            let mut arguments = capture
-                .name("arguments")
-                .expect("arguments capture group")
-                .as_str()
-                .split(' ');
-            log::info!("Predicate: {predicate} -  Arguments: {arguments:?}");
+        let mut indent_level: Option<usize> = None;
+
+        let pragmata = Pragmata::from(query);
+        for pragma in pragmata {
+            let Pragma { predicate, value } = pragma?;
+            log::info!("Pragma: {predicate}, Arguments: {value:?}");
 
             match predicate {
                 "language" => {
-                    if let Some(arg) = arguments.next() {
-                        language = Some(Language::new(arg)?);
+                    if let Some(value) = value {
+                        if language.is_some() {
+                            log::warn!("The #language! pragma has already been set");
+                        }
+
+                        language = Some(Language::new(value)?);
                     } else {
                         return Err(FormatterError::Query(
-                            "The #language! configuration predicate must have a parameter".into(),
+                            "The #language! pragma must have a parameter".into(),
                             None,
                         ));
                     }
                 }
+
                 "indent-level" => {
-                    if let Some(arg) = arguments.next() {
-                        indent_level = arg.parse::<usize>().map_err(|_| {
+                    if let Some(value) = value {
+                        if indent_level.is_some() {
+                            log::warn!("The #indent-level! pragma has already been set");
+                        }
+
+                        indent_level = Some(value.parse().map_err(|_| {
                             FormatterError::Query(
-                                format!(
-                                    "The #indent-level! parameter must be a positive integer, but got '{arg}'"
-                                ),
+                                format!("The #indent-level! pragma expects a positive integer, but got '{value}'"),
                                 None,
                             )
-                        })?;
+                        })?);
                     } else {
                         return Err(FormatterError::Query(
-                            "The #indent-level! configuration predicate must have a parameter"
-                                .into(),
+                            "The #indent-level! pragma must have a parameter".into(),
                             None,
                         ));
                     }
                 }
+
                 _ => {
                     return Err(FormatterError::Query(
-                        format!("Unknown configuration predicate '{predicate}'"),
+                        format!("Unknown pragma '#{predicate}!'"),
```

Comment:
Maybe we could be a little more informative here? Like specifying that the error happened during configuration parsing, not during the parsing of the query for its application to the input? Also, maybe we could provide a list of known pragma?
That may be out of the scope of this PR, though.

Reply:
I've made a _slight_ change to the wording in d24a0e9. I didn't provide a list of known pragmata, largely because I think it ought to be done more generally -- rather than simply using a static string -- and that deserves its own PR. (I'll add an issue for this.)

Reply:
(Submitted Issue #316)